libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java

```diff
@@ -95,6 +96,19 @@ public static AudioCapabilities getCapabilities(Context context) {
     return getCapabilities(context, intent);
   }
 
+  private static AudioCapabilities getExternalSurroundCapabilities(@Nullable Intent intent) {
+    int[] supportedEncodings = EXTERNAL_SURROUND_SOUND_CAPABILITIES;
+
+    if (Util.SDK_INT >= 29) {
+      // Check if DTS Encodings are supported via Direct Playback.
+      int[] dtsTypes = {AudioFormat.ENCODING_DTS, AudioFormat.ENCODING_DTS_HD};
+      supportedEncodings = Util.nullSafeIntegerArrayConcatenation(supportedEncodings,
+          Api29.getDirectPlaybackSupportedEncodings(dtsTypes));
+    }
+    supportedEncodings = Arrays.stream(supportedEncodings).distinct().toArray();
```

Comment:
I'm afraid that we currently cannot use Java 8 methods (`Arrays.stream(int[])`). And do we even need to de-duplication here?

Reply:
Added duplicated encodings check in one more place: 4637d5f832de88efc500274d3c2e7587a4f2c216. Thanks.